npre/umbral.py

```diff
@@ -49,8 +52,16 @@ def __init__(self, curve=curves.secp256k1, g=None):
 
     def kdf(self, ecdata):
         # XXX length
-        for_hash = ec.serialize(ecdata)[1:]  # Remove the first (type) bit
-        return keccak(for_hash).digest()
+        ecdata = ec.serialize(ecdata)[1:]  # Remove the first (type) bit
+
+        # TODO: Handle salt somehow
+        return HKDF(
+            algorithm=hashes.SHA256(),
+            length=32,
```

Comment:
I think, you can just use self.bitsize (then it will work for any curve as intended)